### PR TITLE
Upgrade to do Type Mapping configurable

### PR DESCRIPTION
**Pull request by gcorvera**
Changes to use config files outside of the binary to make it easy to associate vocab values to healthvault datatypes

### CODE REVIEW

File: Microsoft.HealthVault.Fhir/Vocabularies/VocabToHealthVaultDictionaries.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace Microsoft.HealthVault.Fhir.Vocabularies
+{
+    internal class VocabToHealthVaultDictionaries
+    {
+        private static VocabToHealthVaultDictionaries s_instance;
+        private static object _lockInstance = new object();
```
Comment (brandonpollett) [inline Microsoft.HealthVault.Fhir/Vocabularies/VocabToHealthVaultDictionaries.cs]: s_lockInstance

File: Microsoft.HealthVault.Fhir/Vocabularies/VocabToHealthVaultDictionaries.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace Microsoft.HealthVault.Fhir.Vocabularies
```
Comment (brandonpollett) [inline Microsoft.HealthVault.Fhir/Vocabularies/VocabToHealthVaultDictionaries.cs]: I don't think vocabularies is the right place for this. In the context of FHIR these are codes/codeableconcept, so I think moving the namespace would be appropriate.

Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Vocabularies/VocabToHealthVaultDictionaries.cs]: Agree, I am adding those changes to the BP branch.

File: Microsoft.HealthVault.Fhir/Vocabularies/VocabToHealthVaultDictionaries.cs (hunk 1 above)
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Vocabularies/VocabToHealthVaultDictionaries.cs]: Fixed on the BP branch too. Thanks

**Pull request merged**